spiralgo/ecommserver
# Adds JPA classes and REST services

After this PR code merges, when the reviewer runs the project, a `product` table on `ecommdb` will be automatically created by JPA, using the entity class `Product.java`.

**Therefore, the reviewer should check if the empty table was created successfully, first.**

This PR also creates two REST services:

1-  ` /create`: 
 
  **Creates new Product with given values.** 

   **TEST:**
   The reviewer should use `Postman` like this:
   - Create a new `POST` request with the URL: `http://localhost:8080/product/create`
   - Fill the body input (checking JSON as input type) with this Product JSON data:
 ```js
  {
    "name": "ABC-Vertexer 900 GPU",
    "price": 2000,
    "imageUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/5/5f/Utah_teapot_simple_2.png/640px-Utah_teapot_simple_2.png",
    "details": "The most modern technology GPU at its price."
}
```
 

![image](https://user-images.githubusercontent.com/60903744/130568758-fe3b37e2-2ddf-4662-ace0-47338e20e542.png)

***
2-  ` /all`:

 **Lists all Products in the DB.**

**TEST**
 - The reviewer should create a `GET` request with the URL: `http://localhost:8080/product/all`
 - See the result list of all products like this:
 
 
![image](https://user-images.githubusercontent.com/60903744/130569248-8b2c3b6c-882f-4aba-9662-a258c51dcf88.png)

- Then, should see if  the `Product `is inserted into the DB:

![image](https://user-images.githubusercontent.com/60903744/130570186-a9f7995b-8375-4d18-ab81-c584fb9f6661.png)
